unittests/tools/test_drheader_parser.py

```diff
@@ -32,3 +32,10 @@ def test_parse_file_has_many_finding_one_tool3(self):
         findings = parser.get_findings(testfile, Test())
         testfile.close()
         self.assertEqual(11, len(findings))
+
+    def test_parse_file_has_many_finding_multiple_urls(self):
+        testfile = open("unittests/scans/drheader/multiple_urls.json")
+        parser = DrHeaderParser()
+        findings = parser.get_findings(testfile, Test())
+        testfile.close()
+        self.assertEqual(4, len(findings))
```

Comment:
Please add also a validation of endpoints as it is recommended in the documentation: https://documentation.defectdojo.com/contributing/how-to-write-a-parser/#endpoint-validation
Plus some small checks for content of endpoints (e.g. `e.protocol == 'http'` and `e.host == 'example2.com'`

Reply:
Do you want the small checks for content of endpoints in the unittest or rather as an endpoint value? Could you desribe that better and show how to do it?

Reply:
I highly recommend checking other parsers which use endpoints as well (`fgrep -r unsaved_endpoints unittests/tools`). They usually have also tests for endpoints.

But some examples:
```python
unittests/tools/test_gitlab_dast_parser.py-        endpoint = finding.unsaved_endpoints[0]
unittests/tools/test_gitlab_dast_parser.py-        self.assertEqual(str(endpoint), "http://api-server/v1/tree/10")
unittests/tools/test_nikto_parser.py-                endpoint = finding.unsaved_endpoints[0]
unittests/tools/test_nikto_parser.py-                self.assertEqual(8070, endpoint.port)
unittests/tools/test_nikto_parser.py:                self.assertEqual("127.0.0.1", endpoint.host)
unittests/tools/test_nikto_parser.py-                self.assertEqual("examples/servlets/index.html", endpoint.path)
unittests/tools/test_nikto_parser.py-            endpoint = finding.unsaved_endpoints[0]
unittests/tools/test_nikto_parser.py-            self.assertEqual(443, endpoint.port)
unittests/tools/test_nikto_parser.py:            self.assertEqual("64.220.43.153", endpoint.host)
unittests/tools/test_nikto_parser.py-            self.assertIsNone(endpoint.path)
unittests/tools/test_openvas_csv_parser.py-            endpoint = finding.unsaved_endpoints[0]
unittests/tools/test_openvas_csv_parser.py:            self.assertEqual("LOGSRV", endpoint.host)
unittests/tools/test_openvas_csv_parser.py-            self.assertEqual("tcp", endpoint.protocol)
unittests/tools/test_openvas_csv_parser.py-            self.assertEqual(9200, endpoint.port)
```